src/qt/transactionrecord.cpp

```diff
@@ -26,6 +26,17 @@ bool TransactionRecord::showTransaction(const CWalletTx &wtx)
             return false;
         }
     }
+
+    // Suppress OP_RETURN transactions if they did not originate from you.
+    // This is not "very" taxing but necessary since the transaction is in the wallet already.
+    if (!wtx.IsFromMe())
+    {
+        for (auto const& txout : wtx.vout)
+        {
+            if (txout.scriptPubKey == (CScript() << OP_RETURN))
```

Comment:
the logic behind this is that if any tx with multiple txs contains this with no address in scriptpubkey it will display in all wallets. This allows OP_RETURN to be used with data in a transaction still however if its only a OP_RETURN and no valid address we just don't show the transaction.

The gibberish address we experienced and saw is the checksum of null for an address. Other coins experienced similar in the past.